crypto/s2n_rsa.c

```diff
@@ -273,3 +208,60 @@ int s2n_rsa_keys_match(struct s2n_rsa_public_key *pub, struct s2n_rsa_private_ke
 
     return 0;
 }
+
+static int s2n_rsa_key_free(struct s2n_pkey *pkey)
+{
+    struct s2n_rsa_key *rsa_key = &pkey->key.rsa_key;
+
+    RSA_free(rsa_key->rsa);
+    rsa_key->rsa = NULL;
+    
+    return 0;
+}
+
+int s2n_rsa_check_key(const struct s2n_pkey *pkey)
```

Comment:
From the `RSA_check_key()` doc, this can only be used on `RSA` structs with private key data. As such, I think this should be modified to:
```
int s2n_rsa_check_key(const struct s2n_rsa_private_key *rsa_key)
{
    notnull_check(rsa_key->rsa);
    GUARD(RSA_check_key(rsa_key->rsa));
    return 0;
}
```
Thoughts?

Reply:
It's been so long since I've written this code, that I don't remember my rationale for writing it this way. It seems like the way it is written wouldn't work at all because, as you said, RSA_check_key should only work on public keys. I've renamed this function to imply that it is simply doing a check that the public key pkey was initialized appropriately. I *think* that is what I originally intended. It is only called in the client auth validation callbacks which requires that the public key is extracted from the cert. Other calls to RSA_check_key (that are called on private keys) remain as they were.